app/src/main/java/com/isc/hermes/model/Utils/MapPolyline.java

```diff
@@ -58,24 +84,68 @@ public List<Point> getCoordinates(LineString lineString){
-                        PropertyFactory.lineWidth(4f)
-                );
-                style.addLayer(polylineLayer);
+
+                List<Point> polyline1Points = points.get(0);
+                LineString polyline1 = LineString.fromLngLats(polyline1Points);
+
+                List<Point> polyline2Points = points.get(1);
+                LineString polyline2 = LineString.fromLngLats(polyline2Points);
+
+                List<Point> polyline3Points = points.get(2);
+                LineString polyline3 = LineString.fromLngLats(polyline3Points);
+
+
+                List<Feature> features = new ArrayList<>();
+                features.add(Feature.fromGeometry(polyline1));
+                features.add(Feature.fromGeometry(polyline2));
+                features.add(Feature.fromGeometry(polyline3));
+                FeatureCollection featureCollection = FeatureCollection.fromFeatures(features);
+
+                GeoJsonSource polyline1Source = new GeoJsonSource("polyline1-source", featureCollection);
+                GeoJsonSource polyline2Source = new GeoJsonSource("polyline2-source", featureCollection);
+                GeoJsonSource polyline3Source = new GeoJsonSource("polyline3-source", featureCollection);
+                style.addSource(polyline1Source);
+                style.addSource(polyline2Source);
+                style.addSource(polyline3Source);
+
+                if(polyline1Points != null){
+                    LineLayer polyline1Layer = new LineLayer("polyline1-layer", "polyline1-source");
+                    polyline1Layer.setProperties(
+                            PropertyFactory.lineColor(colors.get(0)),
+                            PropertyFactory.lineWidth(4.5f)
+                    );
+                    style.addLayer(polyline1Layer);
+                }
+
+                if(polyline2Points != null){
+                    System.out.println("BOBO");
+                    LineLayer polyline2Layer = new LineLayer("polyline2-layer", "polyline2-source");
+                    polyline2Layer.setProperties(
+                            PropertyFactory.lineColor(colors.get(1)),
+                            PropertyFactory.lineWidth(4.5f)
+                    );
+                    style.addLayer(polyline2Layer);
+                }
+
+                if(polyline3Points != null){
+                    LineLayer polyline3Layer = new LineLayer("polyline3-layer", "polyline3-source");
+                    polyline3Layer.setProperties(
+                            PropertyFactory.lineColor(colors.get(2)),
+                            PropertyFactory.lineWidth(4.5f)
+                    );
+                    style.addLayer(polyline3Layer);
+                }
+
+                polyline1Source.setGeoJson(features.get(0));
+                polyline2Source.setGeoJson(features.get(1));
+                polyline3Source.setGeoJson(features.get(2));
             });
         });
     }
```

Comment:
This method is so long, please refactor it.

Reply:
Done